hibernate-core/src/main/java/org/hibernate/collection/internal/PersistentBag.java

```diff
@@ -137,34 +136,35 @@ public void beforeInitialize(CollectionPersister persister, int anticipatedSize)
 	@Override
 	@SuppressWarnings( "unchecked" )
 	public boolean equalsSnapshot(CollectionPersister persister) throws HibernateException {
-		Type elementType = persister.getElementType();
-		List<Object> sn = (List<Object>) getSnapshot();
+		final Type elementType = persister.getElementType();
+		final List<Object> sn = (List<Object>) getSnapshot();
 		if ( sn.size() != bag.size() ) {
 			return false;
 		}
 
 		// HHH-11032 - Group objects by Type.getHashCode() to reduce the complexity of the search
-		Map<Integer, List<Object>> instanceCountBag = groupByEqualityHash( bag, elementType );
-		Map<Integer, List<Object>> instanceCountSn = groupByEqualityHash( sn, elementType );
-		if ( instanceCountBag.size() != instanceCountSn.size() ) {
+		final Map<Integer, List<Object>> hashToInstancesBag = groupByEqualityHash( bag, elementType );
+		final Map<Integer, List<Object>> HashToInstancesSn = groupByEqualityHash( sn, elementType );
```

Comment:
Minor - but should be `hashToInstancesSn` rather than `HashToInstancesSn`

Reply:
oops, yes, I'll make that change.

Reply:
Oops, yes, I'll make that change.